### ISSUE TITLE
room directory search is case sensitive in synapse 1.5.0

### ISSUE THREAD

**Issue by RichGriese** [matrix-org/synapse#6265]
Hello folks. Originally posed this in Riot Web/Desktop room, and they told me the issue is a "Synapse" issue. I'm a knob so I had to look up what that is, and then come find this room. And with that setup info... Wanted to mention again what seems to me to be something simple to fix. I'm using the riot desktop app. I notice if I go to "explore" and type in a word, that the search is case sensitive. Seems the search should be case insensitive, and would simple fix. Enjoy life!

**Comment by richvdh** [comment]
"Explore" is the room directory search?

afaik it is case-insensitive already. Perhaps you can give an example of what you're searching for, and what you'd expect to find that you're not?

**Comment by RichGriese** [comment]
@richvdh Thanks for the response. Yeah... when you click the link the popup says "Explore rooms
If you can't find the room you're looking for, ask for an invite or Create a new room."

I've build a little test room, and added the items I wanted found with BOTH the capitalization and not, but you can still she how it works "or doesn't" by searching on both "richgriese" and then "RichGriese". It only finds the one that is capitalized. The search searches BOTH the room name and the description.

**Comment by richvdh** [comment]
Ah sorry, this is something that has changed in synapse 1.5. I was testing against 1.4.1.

**Comment by RichGriese** [comment]
@richvdh So it sounds like what you're saying is... That the matrix.org riot server instance uses Synapse as part of it's install... and that if/when they update their instance software that the feature will instantly change?

**Comment by richvdh** [comment]
no, I'm saying that the matrix.org matrix homeserver is using (the forthcoming) synapse 1.5.0 release, where room-directory search is case-sensitive.

I was testing on a different homeserver, which is running synapse 1.4.1, which is why I didn't see the case-sensitive behaviour earlier.

**Comment by RichGriese** [comment]
@richvdh that seems strange. You mean Synapse is moving from being case insensitive to case sensitive on room searches? Is that a bug for the 1.5 release? Or is there a reason to move to these searches being case insensitive?

**Comment by richvdh** [comment]
it was not deliberate afaik. Synapse 1.5 includes a rewrite of the room directory to make it much faster.

**Comment by RichGriese** [comment]
@richvdh That's "good" news to some extent... in that it was not a decision which I would argue with, but more of a "bug" that snuck in as part of attempts to improve something different (speed).

So... now the question is... is there a way to re-class this post as a bug instead of a feature request? when I created it the system presented me with 3 choices, bug, suggestion, and something I forgot. I chose feature request. From what your tell me, I would not consider this a bug for the upcoming / recent(?) 1.5 synapse release.
